src/ar100/Makefile

```diff
@@ -0,0 +1,41 @@
+CROSS_PREFIX=or1k-linux-musl-
+dirs-y += src/generic src/ar100 lib/ar100
+
+CFLAGS		+= -O3
+CFLAGS		+= -fno-builtin
+CFLAGS		+= -fno-pie
+CFLAGS		+= -ffreestanding
+CFLAGS		+= -msfimm -mshftimm -msoft-div -msoft-mul
+CFLAGS		+= -Ilib/ar100
+CFLAGS_klipper.elf := $(CFLAGS) -Wl,-T,src/ar100/ar100.ld -Wl,-Map,out/klipper.map
+CFLAGS_klipper.elf += -Wl,--no-undefined  -Wl,--gc-sections -static
+CFLAGS_klipper.elf += -Wl,--no-dynamic-linker -Wl,--allow-multiple-definition
+
+SFLAGS		 = -nostdinc -Wa,--fatal-warnings -MMD -Wmissing-prototypes -Wall -Wextra
+SFLAGS		+= -Ilib/ar100
+
+# Add source files
+src-y += ar100/main.c ar100/gpio.c ar100/serial.c
+src-y += ar100/util.c ar100/timer.c
+src-y += generic/crc16_ccitt.c generic/timer_irq.c
+src-y += buttons.c tmcuart.c spi_software.c
+src-y += spicmds.c sensor_adxl345.c
```

Comment:
This looks a bit suspect - is there a reason not to enable HAVE_GPIO_SPI and/or HAVE_GPIO_BITBANGING?

Reply:
There is not enough space for the default options. I've instead switched to removing the files that are not needed for this board, is that better?

Reply:
Your latest PR looks fine to me, but I'm curious what "I've instead switched to removing the files that are not needed for this board" means.  What was the change?

Cheers,
-Kevin

Reply:
I'm now using HAVE_GPIO_SPI and HAVE_GPIO_BITBANGING via Kconfig, but since those options are also pulling in some LCD-related files etc. I'm removing unneded files for AR100  instead for of adding using src-y. I've not seen this pattern elsewhere in the code, so it is perhaps not welcome. The third alternative would be to have more fine grained control in the Kconfigs that separates LCD-related files from SPI-realated files.